st2common/st2common/models/api/rule_enforcement.py

```diff
@@ -0,0 +1,93 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import six
+
+from st2common.models.api.base import BaseAPI
+from st2common.models.db.rule_enforcement import RuleEnforcementDB
+from st2common.util import isotime
+
+
+class RuleEnforcementAPI(BaseAPI):
+    model = RuleEnforcementDB
+    schema = {
+        'title': 'RuleEnforcement',
+        'description': 'A specific instance of rule enforcement.',
+        'type': 'object',
+        'properties': {
+            'trigger_instance_id': {
+                'description': 'The unique identifier for the trigger instance ' +
+                               'that flipped the rule.',
+                'type': 'string',
+                'required': True
+            },
+            'execution_id': {
+                'description': 'ID of the action execution that was invoked as a response.',
+                'type': 'string',
+                'required': True
+            },
+            'rule_ref': {
+                'description': 'Reference to the rule object.',
+                'type': 'string',
+                'required': True
+            },
+            'rule_id': {
+                'description': 'ID of the corresponding rule.',
+                'type': 'string',
+                'required': True
+            },
+            'rule_uid': {
+                'description': 'UID of the corresponding rule.',
+                'type': 'string',
+                'required': True
+            },
+            'rule_pack': {
```

Comment:
Because, rbac. User could have pack level permissions and we need to know the pack, the rule in enforcement belongs to. I could have done get_by_ref on rule in rbac resolver and figured out the pack name but this turned out cheaper.